app/com/linkedin/drelephant/spark/fetchers/SparkFetcher.scala

```diff
@@ -100,19 +101,19 @@ class SparkFetcher(fetcherConfigurationData: FetcherConfigurationData)
     )
   }
 
-  private def doFetchSparkApplicationData(analyticJob: AnalyticJob): Future[SparkApplicationData] = {
+  private def doFetchSparkApplicationData(analyticJob: AnalyticJob, fetchFailedTasks: Boolean): Future[SparkApplicationData] = {
     if (shouldProcessLogsLocally) {
       async {
         sparkRestClient.fetchEventLogAndParse(analyticJob.getAppId)
       }
     } else {
-      doFetchDataUsingRestAndLogClients(analyticJob)
+      doFetchDataUsingRestAndLogClients(analyticJob, fetchFailedTasks)
     }
   }
 
-  private def doFetchDataUsingRestAndLogClients(analyticJob: AnalyticJob): Future[SparkApplicationData] = async {
+  private def doFetchDataUsingRestAndLogClients(analyticJob: AnalyticJob, fetchFailedTasks: Boolean): Future[SparkApplicationData] = async {
     val appId = analyticJob.getAppId
-    val restDerivedData = await(sparkRestClient.fetchData(appId, eventLogSource == EventLogSource.Rest))
+    val restDerivedData = await(sparkRestClient.fetchData(appId, eventLogSource == EventLogSource.Rest, fetchFailedTasks))
```

Comment:
I noticed you were passing fetchFailedTasks in every method prior to this? Can't you compute it directly over here?

Reply:
Done.